### ISSUE TITLE
Allow to disable cache in a task, in config

### ISSUE THREAD

**Issue by soloam** [Flexget/Flexget#2667]
### Expected behaviour:

It would be nice if we would be able to a a no_cache: yes to a task and that would disable cache in that task. Some tasks the cache can make strange results because we my depend on updated results on every single run. 

### Actual behaviour:

We are able to disable the cache, but for that we need to run the task with --no-cache, that is not possible if we are running the tasks with daemon! 

Thank you

**Comment by paranoidi** [comment]
People will missuse it and hammer services causing flexget to be blacklisted. That is why it's only available when testing from cli. When are you getting strange results?

**Comment by soloam** [comment]
Yes I know that people using it on all tasks would be potentially harmful! But who what's to do that no matter what just run all tasks with cron and the --no-cache.

In my case I have a group of tasks that run around once a day that test some assumptions of my configuration, like having all movies of imdb in my lists, and removing from the lists what I remove from imdb, removing movies with incomplete data, basically maintenence tasks! This tasks depend on the state of the lists of imdb and trakt! And the cache can make strange results!

Also, but this I have to confirm, I use the text plug-in to read info from a file! I had some weird results when modifying the file in an task, and read it in the next one! I do this to capture data with the exec, and writing the file to use it with text in the next one! And I was only able to make this work if every task creates a new file name! I assumed that it was a cache problem, but I never tested further to confirm this!

**Comment by paranoidi** [comment]
Maybe we could do a plugin that clears/removes cache at specific time(s). It would be cumbersome to use in a harmful way.

**Comment by soloam** [comment]
Or a plugin that allows me to know that every x hours the cache will be cleared for sure! I would point that tasks to that hours! Like clear cache ever 6 am for example...

**Comment by BrutuZ** [comment]
I thought FlexGet cache only lasted for 2h such as in
https://github.com/Flexget/Flexget/blob/bf25d23599206df2e8639863aaa0907a05dd0092/flexget/components/trakt/trakt_list.py#L504

**Comment by soloam** [comment]
Probably yes! But what I was referring is knowing exactly that when a task is running that the data is not cached! 2 hours is very abstract, because I don't know when 2 hours ago was!

I don't care if the cache is 2 hours or 12! I only would like a guaranty that at that given time the values would not be cached! I know I can do a cron to that! But I would like keep things in the daemon!

**Comment by evoblicec** [comment]
You can try to use `Schedule` instead of `interval` in your `Schedules` configuration. 

https://flexget.com/Plugins/Daemon/scheduler

With this you can specify when tasks run exactly and take the 2 hours of cache into account

**Comment by soloam** [comment]
That's not a 100% reliable... The task could had been delayed for a number of reasons, queue, lost connections, you name it! That would deviate the 2 hours window!

A plug-in is the most reliable way! At the moment I'm running at request with the --no-cache, but I would like to automate it! My best approach is cron, but I would like to avoid that!

**Comment by evoblicec** [comment]
Sorry I don’t fully understand your problem.

If you run your group of task only once a day what is the matter with the 2hours of cache?

Can you describe the high level logic of what you want to achieve?

**Comment by soloam** [comment]
I run this maintenance tasks once a day! One of the tasks that it does is, if I remove a item from the the imdb list, remove it from the movie list! This is one that had given problems in the past because of cache! Items beings removed because the cache

**Comment by evoblicec** [comment]
> This is one that had given problems in the past because of cache! Items beings removed because the cache

Can you explain further?

If I breakdown the possibilities when you hit the cache and not the real list:

- A)No change to the list, cached version is the same, no impact 
- B) A movie was added to the list since last update therefore is not present in the cache, this new entry will not be processed 
- C) B movie was deleted from the list since last update but is still present in the cache, therefore the entry will not be deleted

In conclusion cached could only delay a deletion from Movie List.

Do you have multiples tasks that are modifying the same list?

Can you provide your YAML configuration to further help you?

**Comment by soloam** [comment]
This issue is complicated to solve due to the way that cache is implemented at the moment, I can close this until we have more time to think in this! Issue #2823 seems a different and should be analysed independently.
